src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/PointsToAnalysis/PointsToAnalysis.PointsToDataFlowOperationVisitor.cs

```diff
@@ -951,12 +998,28 @@ private PointsToAbstractValue GetValueBasedOnInstanceOrReferenceValue(IOperation
             public override PointsToAbstractValue VisitFieldReference(IFieldReferenceOperation operation, object argument)
             {
                 var value = base.VisitFieldReference(operation, argument);
+
+                // "class SomeType { static readonly SomeType Empty; }"
+                if (IsSpecialEmptyMember(operation.Field) &&
+                    value.NullState != NullAbstractValue.Null)
```

Comment:
>value.NullState != NullAbstractValue.Null [](start = 20, length = 41)

Why check for != Null, instead of != NotNull?

Null => definitely null?

Reply:
Just protecting against the case where the readonly member is assigned null, though not very likely. I don't think it will matter right now as we don't analyze readonly fields or properties.